x-pack/plugins/infra/public/containers/logs/log_position/log_position_state.ts

```diff
@@ -79,26 +114,94 @@ export const useLogPositionState: () => LogPositionStateParams & LogPositionCall
+        newDateRange.startDateExpression || dateRange.startDateExpression;
+      const nextEndDateExpression = newDateRange.endDateExpression || dateRange.endDateExpression;
+
+      if (!isValidDatemath(nextStartDateExpression) || !isValidDatemath(nextEndDateExpression)) {
+        return;
+      }
+
+      // Dates are valid, so the function cannot return `null`
+      const nextStartTimestamp = datemathToEpochMillis(nextStartDateExpression)!;
+      const nextEndTimestamp = datemathToEpochMillis(nextEndDateExpression, 'up')!;
+
+      // Reset the target position if it doesn't fall within the new range.
+      if (
+        targetPosition &&
+        (nextStartTimestamp > targetPosition.time || nextEndTimestamp < targetPosition.time)
+      ) {
+        jumpToTargetPosition(null);
+      }
+
+      setDateRange(previousDateRange => {
+        return {
+          startDateExpression:
+            newDateRange.startDateExpression || previousDateRange.startDateExpression,
+          endDateExpression: newDateRange.endDateExpression || previousDateRange.endDateExpression,
+        };
+      });
+      updateTimestamps();
+    },
+    [dateRange, targetPosition, updateTimestamps]
+  );
+
+  // `lastUpdate` needs to be a dependency for the timestamps.
+  // ESLint complains it's unnecessary, but we know better.
+  /* eslint-disable react-hooks/exhaustive-deps */
+  const startTimestamp = useMemo(() => datemathToEpochMillis(dateRange.startDateExpression), [
+    dateRange.startDateExpression,
+    timestampsLastUpdate,
+  ]);
+
+  // endTimestamp needs to be synced to `now` to allow auto-streaming
+  const endTimestampDep =
+    dateRange.endDateExpression === 'now' ? Date.now() : dateRange.endDateExpression;
+  const endTimestamp = useMemo(() => datemathToEpochMillis(dateRange.endDateExpression, 'up'), [
+    endTimestampDep,
+    timestampsLastUpdate,
+  ]);
+  /* eslint-enable react-hooks/exhaustive-deps */
+
   const state = {
+    isInitialized,
     targetPosition,
-    isAutoReloading,
+    isStreaming,
     firstVisiblePosition: startKey,
     pagesBeforeStart,
     pagesAfterEnd,
     visibleMidpoint,
     visibleMidpointTime: visibleMidpoint ? visibleMidpoint.time : null,
     visibleTimeInterval,
+    ...dateRange,
+    startTimestamp,
+    endTimestamp,
+    timestampsLastUpdate,
   };
 
   const callbacks = {
+    initialize,
     jumpToTargetPosition,
     jumpToTargetPositionTime: useCallback(
       (time: number) => jumpToTargetPosition({ tiebreaker: 0, time }),
       [jumpToTargetPosition]
     ),
     reportVisiblePositions,
-    startLiveStreaming: useCallback(() => setIsAutoReloading(true), [setIsAutoReloading]),
-    stopLiveStreaming: useCallback(() => setIsAutoReloading(false), [setIsAutoReloading]),
+    startLiveStreaming: useCallback(() => {
+      setIsStreaming(true);
+      jumpToTargetPosition(null);
+      updateDateRange({ startDateExpression: 'now-1d', endDateExpression: 'now' });
```

Comment:
Just out of curiosity, how was the value `now-1d` chosen?

Reply:
I don't have the background for that. @katrin-freihofner do you know?